### ISSUE TITLE
Incorrect branch coverage when targeting multiple frameworks

### ISSUE THREAD

**Issue by Jericho** [OpenCover/opencover#931]
### My Framework

- [ ] .NET 2
- [ ] .NET 3.5
- [ ] .NET 4
- [ ] .NET 4.5
- [ ] .NET 4.6
- [x] .NET 4.6.1
- [ ] .NET 4.6.2
- [ ] .NET 4.7
- [ ] .NET 4.7.1
- [x] .NET 4.7.2
- [ ] .NET Core 1.0.0
- [x] .NET Core 2.0.0
- [ ] Something else

### My Environment

- [ ] Windows 7 or below (not truly supported due to EOL)
- [ ] Windows 8
- [ ] Windows 8.1
- [x] Windows 10
- [ ] Windows 10 IoT Core
- [ ] Windows Server 2012
- [ ] Windows Server 2012 R2
- [ ] Windows Server 2016

### Actual Behavior

I use OpenCover version 4.7.922 to generate coverage.xml and I use ReportGenerator version 4.4.0 to convert the XML file into a HTML report but something seems odd in the generated report. I initially thought the problem was with ReportGenerator so I raised an issue in their repo but @danielpalme seems to think OpenCover is the culprit.

In short, the situation is that I get "0 of 2 branches" covered when in fact I have unit tests covering the two scenarios. During my investigation of the situation I realized that this problem is most likely due to the fact that my C# library targets multiple frameworks and my unit testing projects also targets multiple frameworks. If I remove the multi-targeting, the report correctly reports that I am covering all branches. You can read more details in this [issue](https://github.com/danielpalme/ReportGenerator/issues/308), including screenshots to demonstrate the problem, C# code sample, sample `coverage.xml` file and also link to a public GitHub repo that will allow you to easily reproduce the situation.

**Comment by Jericho** [comment]
@sawilde You closed this issue but did you investigate the problem I was reporting? Were you able to look at the sample XML file I provided and/or look at the public repo I created to reproduce the problem?

**Comment by sawilde** [comment]
I couldn't get your sample to build and so closed it since it had been open for so long

**Comment by Jericho** [comment]
What is it that didn't work? any error? Anything I can do to help?

- You can read the discussion I had with @danielpalme [here](https://github.com/danielpalme/ReportGenerator/issues/308) to help you understand how I came to the conclusion that the issue seems to be related to unit testing multiple frameworks. Maybe it can help you create your own repro if the one I provided doesn't work for you.
- I have provided a [sample coverage file](https://github.com/danielpalme/ReportGenerator/files/4041743/coverage.xml.txt), maybe it can help make sense of the situation
- I provided what I thought were simple steps to repro but please let me know if it's not clear:
  - Clone the [repo](https://github.com/Jericho/WeirdCoverageRepro)
  - from a Powershell prompt run `.\build --bootstrap` (one time only)
  - From then on, you can execute `.\build -t coverage` to automatically clean previous artifacts, restore nuget packages, build, run unit tests, generate coverage xml, convert xml to html report and open the report in your default browser.

I know it's been open for a long time and I have been patiently waiting for someone to look at it, but please don't dismiss it simply because it's been open for a long time. I invested a lot of time to investigate this issue and I attempted to provide all the info you need to repro including sample coverage XML file and sample C# project. Please tell me what else you need.

**Comment by sawilde** [comment]
@Jericho the suspicious thing is that all three variations of the method have the same value for the MetadataToken 

What happens if you run each test assembly individually in separate sessions rather than in the one session?
